scroll-search.lic

```diff
@@ -0,0 +1,533 @@
+	resave_container(container, scroll_info, false)
+
+	echo_message("SCROLL WAS TRANSFERRED!")
+  end
+
+  def reorder_container(container, scroll_info, wasRemoved)
+	parameter = "C #{scroll_info.get_type}"
+	scrolls = rummage(parameter, get_noun(container))
+	
+	echo("Scroll count of those matching the type (#{scroll_info.get_type}) is: #{scrolls.length}.") if $debug_mode
+	
+	scrolls_in_container = @owned_scrolls[container]
+	scrolls_in_container.each do |scroll_information|
+	
+	  if scroll_information.get_type != scroll_info.get_type
+		next
+	  end
+		
+	  if scroll_information.index == scroll_info.index
+		next
+	  end
+		
+	  if wasRemoved
+		if scroll_information.container_index > scroll_info.container_index && scroll_information.container_index > 0
+		  new_index = scroll_information.container_index - 1
+		  new_ordinal = $ORDINALS[new_index]
+		else
+		  new_index = scroll_information.container_index
+		  new_ordinal = $ORDINALS[new_index]
+		end
+	  else
+		new_index = scroll_information.container_index + 1
+		new_ordinal = $ORDINALS[new_index]
+	  end
+		
+	  echo("New ordinal and index for scroll (#{scroll_information.spell}) is: #{new_index} - #{new_ordinal}.") if $debug_mode
+		
+	  scroll_information.set_ordinal(new_ordinal)
+	  scroll_information.set_container_index(new_index)
+	end
+  end
+
+  def resave_container(container, scroll_info, remove)
+  
+	# if wasRemoved is true, we remove the scroll from
+	# the container. If not, we simply add it to the container specified.
+	
+	if remove
+	  scrolls_in_container = @owned_scrolls[scroll_info.container]
+	  removed = scrolls_in_container.delete(scroll_info)
+	  if removed
+		echo("Deleting scroll #{removed.spell} from container #{scroll_info.container}") if $debug_mode
+		# Re-save owned scrolls listing
+		@owned_scrolls[scroll_info.container] = scrolls_in_container
+			
+		# Re-save user variables
+		UserVars.owned_scrolls = @owned_scrolls
+	  else
+		echo_error("COULD NOT TRANSFER SCROLL TO NEW CONTAINER")
+		exit
+	  end
+	else
+	  echo ("Attempting to add scroll #{scroll_info.spell} to container #{container}.") if $debug_mode
+		
+	  my_scroll_list = []
+
+	  if @owned_scrolls.key? (container)
+		my_scroll_list = @owned_scrolls[container]
+	  end
+
+	  my_scroll_list.push(scroll_info)
+	  if my_scroll_list.length > 0
+		echo("Adding scroll #{scroll_info.spell} to container #{container}. Length of array is #{my_scroll_list.length}") if $debug_mode
+		# Re-save owned scrolls listing
+		@owned_scrolls[container] = my_scroll_list
+			
+		# Re-save user variables
+		UserVars.owned_scrolls = @owned_scrolls
+			
+		# List scrolls
+		my_scroll_list.each do |scroll_array|
+		  echo("Scrolls in container is #{scroll_array.spell}") if $debug_mode
+		end
+			
+		# List everything
+		@owned_scrolls.each do |container, scroll_array|
+		  scroll_array.each do |scroll_information|
+			echo("Container #{container} | #{scroll_information.spell}") if $debug_mode
+		  end
+		end
+	  else
```

Comment:
I recommend checking this case first (e.g. `unless removed`) and then, due to the exit, you do not need to have an else.